docs/query-layer/architecture.md

```
@@ -0,0 +1,275 @@
+id: 019f53ab-e807-7913-bef7-350f780b393c
+slug: specs/query-layer/architecture
+title: "Query layer — architecture and maintainability"
+type: spec
+status: active
+priority: high
+tags: [graphql, query-layer, spec]
+---
+
+# Architecture
+
+### Placement: `graphql` feature on the core crate
```

markdownlint-cli2 (0.22.1)

[warning] Heading levels should only increment by one level at a time
Expected: h2; Actual: h3

(MD001, heading-increment)